src/compiler/checker.ts

```diff
@@ -5589,15 +5729,67 @@ namespace ts {
         function getTypeFromTypeLiteralOrFunctionOrConstructorTypeNode(node: Node, aliasSymbol?: Symbol, aliasTypeArguments?: Type[]): Type {
             const links = getNodeLinks(node);
             if (!links.resolvedType) {
-                // Deferred resolution of members is handled by resolveObjectTypeMembers
-                const type = createObjectType(TypeFlags.Anonymous, node.symbol);
+                const isSpread = (node.kind === SyntaxKind.TypeLiteral &&
+                                  find((node as TypeLiteralNode).members, elt => elt.kind === SyntaxKind.SpreadTypeElement));
+                let type: ObjectType;

```

Comment:
You never use `type` in the first branch, so do you really need to declare it up here?